File: tests/nodejs/cases/case_100__ledger.js
/**
 * Task case 100: _ledger
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case100Ledger(input) {
  // Task: internal ledger records and prepare transport-ready payload.
  const taskName = '_ledger';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'ledger',
    stage: '',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // SAFE: payload remains local; only primitive checksum string is returned.
  return payload.checksum;
}

module.exports = {
  case100Ledger,
};

File: tests/python/cases/case_100__ledger.py
"""
Task case 100:  ledger payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_100__ledger(input_data):
    # Task:  ledger records and prepare transport-ready payload.
    task_name = '_ledger'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'ledger',
        'stage': '',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # SAFE: payload remains local; only primitive checksum string is returned.
    return payload['checksum']
